cdap-master/src/main/java/co/cask/cdap/data/tools/PFSUpgrader.java

```diff
@@ -149,8 +153,64 @@ public void apply() throws Exception {
         }
       }
     });
+  }
+
+  // need to add 'core' as a module dependency for any modules that currently depend on partitionedFileSet or
+  // timePartitionedFileSet as a resolution to: https://issues.cask.co/browse/CDAP-3030
+  private void upgradeModulesDependingOnPfs() throws Exception {
+    final DatasetTypeMDS dsTypeMDS;
+    try {
+      dsTypeMDS = new DatasetMetaTableUtil(dsFramework).getTypeMetaTable();
+    } catch (Exception e) {
+      LOG.error("Failed to access Datasets types meta table.");
+      throw e;
+    }
+
+    TransactionExecutor executor = executorFactory.createExecutor(ImmutableList.of((TransactionAware) dsTypeMDS));
+    executor.execute(new TransactionExecutor.Subroutine() {
+      @Override
+      public void apply() throws Exception {
+        MDSKey key = new MDSKey.Builder().add(DatasetTypeMDS.MODULES_PREFIX).build();
+        Map<MDSKey, DatasetModuleMeta> dsSpecs = dsTypeMDS.listKV(key, DatasetModuleMeta.class);
+
+        for (Map.Entry<MDSKey, DatasetModuleMeta> entry : dsSpecs.entrySet()) {
+          DatasetModuleMeta moduleMeta = entry.getValue();
+          if (!needsConverting(moduleMeta)) {
+            continue;
+          }
+          LOG.info("Migrating dataset module meta: {}", moduleMeta);
+          DatasetModuleMeta migratedModuleMeta = migrateDatasetModuleMeta(moduleMeta);
+          dsTypeMDS.write(entry.getKey(), migratedModuleMeta);
+        }
+      }
+    });
+  }
 
+  @VisibleForTesting
+  boolean needsConverting(DatasetModuleMeta moduleMeta) {
+    // if 'core' is already a declared dependency, no need to upgrade the module meta
+    List<String> usesModules = moduleMeta.getUsesModules();
+    if (usesModules.contains("core")) {
+      return false;
+    }
+    return usesModules.contains("partitionedFileSet") || usesModules.contains("timePartitionedFileSet");
```

Comment:
It is looking for a module name, not a dataset type name. The module name is unique